package.json

```diff
@@ -20,66 +21,82 @@
     "angular-messages": "npm:@xlts.dev/angular-messages@^1.9.1",
     "angular-sanitize": "npm:@xlts.dev/angular-sanitize@^1.9.1",
-    "firebase": "^9.0.0",
-    "rxfire": "^6.0.0",
-    "rxjs": "^6.6.7",
-    "tslib": "^2.3.1",
-    "zone.js": "~0.11.4"
+    "firebase": "^9.16.0",
+    "ngx-toastr": "^16.0.2",
+    "rxfire": "^6.0.3",
+    "rxjs": "^7.8.0",
+    "tslib": "^2.4.1",
+    "zone.js": "~0.12.0"
   },
   "devDependencies": {
-    "@angular-devkit/architect": "^0.1202.7",
-    "@angular-devkit/build-angular": "12.2.8",
-    "@angular/cli": "12.2.8",
-    "@angular/compiler-cli": "12.2.8",
-    "@angular/language-service": "12.2.8",
+    "@angular-devkit/architect": "^0.1501.2",
+    "@angular-devkit/build-angular": "~15.1.1",
+    "@angular/cli": "15.1.1",
+    "@angular/compiler-cli": "15.1.1",
+    "@angular-eslint/builder": "15.2.0",
+    "@angular-eslint/eslint-plugin": "15.2.0",
+    "@angular-eslint/eslint-plugin-template": "15.2.0",
+    "@angular-eslint/schematics": "15.2.0",
+    "@angular-eslint/template-parser": "15.2.0",
+    "@angular/language-service": "15.1.1",
     "@playwright/test": "^1.30.0",
-    "@types/gulp": "^4.0.9",
-    "@types/node": "^14.17.20",
+    "@types/angular": "^1.8.4",
+    "@types/angular-animate": "^1.5.11",
+    "@types/angular-aria": "^1.7.2",
+    "@types/angular-material": "^1.1.73",
+    "@types/angular-sanitize": "^1.8.1",
+    "@types/gulp": "^4.0.10",
+    "@types/node": "^18.11.18",
+    "@typescript-eslint/eslint-plugin": "5.49.0",
+    "@typescript-eslint/parser": "5.49.0",
     "codelyzer": "^6.0.2",
     "debug": "^4.3.4",
-    "firebase-tools": "^9.19.0",
+    "eslint": "~8.32.0",
+    "eslint-config-prettier": "8.6.0",
+    "eslint-import-resolver-typescript": "^2.5.0",
+    "eslint-plugin-ban": "^1.6.0",
+    "eslint-plugin-import-order": "^2.1.4",
+    "eslint-plugin-prettier": "^4.2.1",
+    "eslint-plugin-rxjs": "^5.0.2",
+    "firebase-tools": "^11.21.0",
     "fuzzy": "^0.1.3",
     "gulp": "~4.0.2",
-    "gulp-angular-templatecache": "~3.0.0",
+    "gulp-angular-templatecache": "~3.0.1",
     "gulp-htmlmin": "~5.0.1",
-    "husky": "~4.3.8",
-    "inquirer": "^6.2.2",
-    "inquirer-autocomplete-prompt": "^1.0.1",
-    "jsonc-parser": "^3.0.0",
-    "lint-staged": "^11.1.2",
-    "npm-check": "^5.9.2",
-    "open": "^7.0.3",
-    "prettier": "^2.4.1",
+    "husky": "~8.0.3",
+    "inquirer": "^9.1.4",
+    "inquirer-autocomplete-prompt": "^3.0.0",
+    "jsonc-parser": "^3.2.0",
+    "lint-staged": "^13.1.0",
+    "npm-check": "^6.0.1",
+    "open": "^8.4.0",
+    "prettier": "^2.8.3",
     "rxjs-tslint-rules": "^4.34.8",
-    "source-map-explorer": "^2.5.2",
-    "ts-node": "^10.2.1",
-    "tslint": "~6.1.3",
-    "tslint-config-prettier": "1.18.0",
+    "source-map-explorer": "^2.5.3",
+    "ts-node": "^10.9.1",
     "tsutils": "^3.21.0",
-    "typescript": "~4.2.4"
+    "typescript": "~4.9.4"
   },
   "husky": {
     "hooks": {
```

Comment:
The engines below is set to npm v6 only. But you've updated the package-lock.json file using npm v8. We either need to update this PR to use only npm v6, or update the engines field below if we're sure that the updates to dependencies here all work with npm v8. If that's the case, we should also update the `.npmrc` file to at least 16 and verify that works.

Reply:
I set the `engine-strict` flag in `.npmrc`. I updated the npm and node versions in the `package.json` engines: `npm v9`, `node>=18`. I tested if the package installation, start and testing of the application works correctly. Everything looks fine 🎉

Reply:
:tada:

Personally, I wouldn't go as far as enabling `engine-strict`. The rest lgtm :+1:

Reply:
Disabled `engine-stric` flag